src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -108,18 +108,28 @@ def grant(self) -> ApiGrant:
                 .get(code=self.code)
             )
         except ApiGrant.DoesNotExist:
-            raise APIUnauthorized("Could not find grant")
+            raise SentryAppIntegratorError(
+                "Could not find grant for given code",
+                webhook_context={"code": self.code[:4]},
+                status_code=404,
+            )
 
     @property
     def application(self) -> ApiApplication:
         try:
             return self.grant.application
         except ApiApplication.DoesNotExist:
-            raise APIUnauthorized("Could not find application")
+            raise SentryAppSentryError(
+                "Could not find application from grant",
+                webhook_context={"code": self.code[:4], "grant_id": self.grant.id},
+            )
```

Comment:
`APIUnauthorized` has status code 401

```suggestion
            raise SentryAppSentryError(
                "Could not find application from grant",
                webhook_context={"code": self.code[:4], "grant_id": self.grant.id},
                status_code=401
            )
```